Cannot load library /tmp/_MEIUOOPug/PySide6/Qt/plugins/platforms/libqxcb.so

Thanks for keeping up with this project, I'm excited to see how it will work and keep the usdb community alive.
Unfortunately, I got the mentioned error while executing the 0.1.0 release file on Debian sid:
"Cannot load library /tmp/_MEIUOOPug/PySide6/Qt/plugins/platforms/libqxcb.so"

Installing following packages due to recommendations at https://stackoverflow.com/questions/68036484/qt6-qt-qpa-plugin-could-not-load-the-qt-platform-plugin-xcb-in-even-thou/72326758#72326758

```
sudo apt install libxcb-xinerama0
sudo apt install libxcb-cursor0
```

has different effects on execution of the clone and the release:

## Release

still has an error with some missing package

```
QT_DEBUG_PLUGINS="1" ./USDB_Syncer-0.1.0-Linux
```

turns into the following error

```
#....
qt.core.plugin.factoryloader: Got keys from plugin meta data QList("eglfs")
qt.core.plugin.factoryloader: checking directory path "/home/ubuntu/Downloads/platforms" ...
qt.core.library: "/tmp/_MEIeniTT4/PySide6/Qt/plugins/platforms/libqxcb.so" cannot load: Cannot load library /tmp/_MEIeniTT4/PySide6/Qt/plugins/platforms/libqxcb.so: (libQt6XcbQpa.so.6: cannot open shared object file: No such file or directory)
qt.core.plugin.loader: QLibraryPrivate::loadPlugin failed on "/tmp/_MEIeniTT4/PySide6/Qt/plugins/platforms/libqxcb.so" : "Cannot load library /tmp/_MEIeniTT4/PySide6/Qt/plugins/platforms/libqxcb.so: (libQt6XcbQpa.so.6: cannot open shared object file: No such file or directory)"
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, wayland, wayland-egl, vnc, vkkhrdisplay, offscreen, minimalegl, minimal, linuxfb, eglfs.

Aborted (core dumped)
```

## clone

seems to have no more missing packages but 

```
usdb_syncer
```

now runs into

```
qt.dbus.integration: Could not connect "org.freedesktop.IBus" to globalEngineChanged(QString)
2023-05-21 22:20:17 [ERROR] Uncaught exception:
    Traceback (most recent call last):
      File "/home/ubuntu/git/usdb_syncer/venv/bin/usdb_syncer", line 8, in <module>
    sys.exit(cli_entry())
             ^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/main.py", line 31, in cli_entry
    main()
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/gui/mw.py", line 309, in main
    _load_main_window(mw)
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/gui/mw.py", line 318, in _load_main_window
    songs = get_all_song_data(False)
            ^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/song_list_fetcher.py", line 23, in get_all_song_data
    songs = get_available_songs(force_reload)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/song_list_fetcher.py", line 49, in get_available_songs
    songs += get_usdb_available_songs(max_skip_id, session=session)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/usdb_scraper.py", line 333, in get_usdb_available_songs
    html = get_usdb_page(
           ^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/usdb_scraper.py", line 231, in get_usdb_page
    return page()
           ^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/usdb_scraper.py", line 222, in page
    session or SessionManager.session(),
               ^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/usdb_scraper.py", line 78, in session
    cls._session = new_session_with_cookies(settings.get_browser())
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/usdb_scraper.py", line 64, in new_session_with_cookies
    if cookies := browser.cookies():
                  ^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/src/usdb_syncer/settings.py", line 224, in cookies
    return function(domain_name=Usdb.DOMAIN)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/venv/lib/python3.11/site-packages/browser_cookie3/__init__.py", line 1028, in chrome
    return Chrome(cookie_file, domain_name, key_file).load()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/venv/lib/python3.11/site-packages/browser_cookie3/__init__.py", line 572, in __init__
    super().__init__(browser='Chrome', cookie_file=cookie_file, domain_name=domain_name, key_file=key_file, **args)
      File "/home/ubuntu/git/usdb_syncer/venv/lib/python3.11/site-packages/browser_cookie3/__init__.py", line 381, in __init__
    self.__add_key_and_cookie_file(**kwargs)
      File "/home/ubuntu/git/usdb_syncer/venv/lib/python3.11/site-packages/browser_cookie3/__init__.py", line 394, in __add_key_and_cookie_file
    password = _LinuxPasswordManager(USE_DBUS_LINUX).get_password(os_crypt_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/ubuntu/git/usdb_syncer/venv/lib/python3.11/site-packages/browser_cookie3/__init__.py", line 226, in get_password
    return self.__methods_map.get('kwallet')(os_crypt_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: _LinuxPasswordManager.__get_kdewallet_password_jeepney() missing 1 required positional argument: 'key'
```

which looks like #136

It's quite late now, I give up for today, but try to follow up in the next days.

Tried following installs (to resolve the missing packages in the Release):

suggested by https://packages.ubuntu.com/lunar/libxcb-util0-dev:

```
sudo apt install libxcb-util-dev
```

suggested by https://github.com/Nuitka/Nuitka/issues/2138

```
sudo apt install -y \
            libfontconfig1-dev libfreetype6-dev \
            libx11-dev libx11-xcb-dev libxext-dev libxfixes-dev \
            libxi-dev libxrender-dev \
            libxkbcommon-dev libxkbcommon-x11-dev libatspi2.0-dev \
            libopengl0 '^libxcb.*-dev'
```

and suggested by https://stackoverflow.com/questions/68036484/qt6-qt-qpa-plugin-could-not-load-the-qt-platform-plugin-xcb-in-even-thou

```
sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev
```

**BUT** the last Debug output with `libQt6XcbQpa.so.6: cannot open shared object file: No such file or directory` did not change

Then following https://stackoverflow.com/a/74719383 I tried

```
export QT_QPA_PLATFORM=offscreen
```

which turned the output of `./USDB_Syncer-0.1.0-Linux` into

```
qt.dbus.integration: Could not connect "org.freedesktop.IBus" to globalEngineChanged(QString)
2023-05-22 20:20:28 [ERROR] Uncaught exception:
    Traceback (most recent call last):
      File "usdb_syncer/main.py", line 36, in <module>
      File "usdb_syncer/main.py", line 31, in cli_entry
      File "usdb_syncer/gui/mw.py", line 309, in main
      File "usdb_syncer/gui/mw.py", line 318, in _load_main_window
      File "usdb_syncer/song_list_fetcher.py", line 23, in get_all_song_data
      File "usdb_syncer/song_list_fetcher.py", line 49, in get_available_songs
      File "usdb_syncer/usdb_scraper.py", line 333, in get_usdb_available_songs
      File "usdb_syncer/usdb_scraper.py", line 231, in get_usdb_page
      File "usdb_syncer/usdb_scraper.py", line 222, in page
      File "usdb_syncer/usdb_scraper.py", line 78, in session
      File "usdb_syncer/usdb_scraper.py", line 64, in new_session_with_cookies
      File "usdb_syncer/settings.py", line 224, in cookies
      File "browser_cookie3/__init__.py", line 973, in chrome
      File "browser_cookie3/__init__.py", line 516, in __init__
      File "browser_cookie3/__init__.py", line 331, in __init__
      File "browser_cookie3/__init__.py", line 344, in __add_key_and_cookie_file
      File "browser_cookie3/__init__.py", line 238, in get_password
    TypeError: _LinuxPasswordManager.__get_kdewallet_password_jeepney() missing 1 required positional argument: 'key'
[12852] Failed to execute script 'main' due to unhandled exception!
```

which now looks pretty much identical to #136 

while the Output of the clone remained unchanged

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -47,6 +47,10 @@ jobs:
         run: generate_pyside_files
       - name: Scrape USDB song list to ship with the bundle
         run: generate_song_list_json -t 'data/song_list.json' -u '${{ secrets.USDB_USER }}' -p '${{ secrets.USDB_PASSWORD }}'
+      - name: add dependencies for xcb support
+        if: matrix.TARGET == 'Linux'
+        run: |
+          sudo apt install libxcb-cursor0
       - name: Build with pyinstaller for ${{matrix.os}}
         run: pyinstaller -n 'USDB_Syncer-${{github.ref_name}}-${{matrix.TARGET}}' ${{matrix.PYINSTALLER_ARGS}} src/usdb_syncer/main.py
       - name: Create DMG

--- a/README.md
+++ b/README.md
@@ -6,16 +6,54 @@
 [![Release](https://github.com/bohning/usdb_syncer/actions/workflows/release.yaml/badge.svg)](https://github.com/bohning/usdb_syncer/actions/workflows/release.yaml)
 [![tox](https://github.com/bohning/usdb_syncer/actions/workflows/tox.yaml/badge.svg)](https://github.com/bohning/usdb_syncer/actions/workflows/tox.yaml)
 
-## Install (Linux)
+## Linux Distributions
 
-requires extra packages on Linux
+Linux Build are generated on `Ubuntu:latest`, should run on `Ubuntu >=23.04`
 
-``` bash
-apt update
-apt install libdbus-1-3
-```
+known requirements:
+
+- package `glibc >= 2.35`
+
+  Therefore the following table (based on https://pkgs.org/search/?q=glibc, 8.7.2023) summarizes different Linux Distributions for having greater or equal version of `glibc`. For (likely) supported distributions the minimum OS Version is given that has a required glibc version. For (likely) unsupported distributions the recent highest Versions (if known) of Linux Distributions with its highest glibc version is given:
+
+  |                    | OS                  | OS Version       | glibc     |
+  |:------------------:|:-------------------:|:----------------:|:---------:|
+  | :x:                | AlmaLinux           | 9                | 2.34 |
+  | :x:                | ALT Linux           | P10              | 2.32 |
+  | :x:                | Amazon Linux        | 2                | 2.26 |
+  | :white_check_mark: | Arch Linux          |                  | 2.37 |
+  | :x:                | CentOS              | 9                | 2.34 |
+  | :x:                | Enterprise Linux    | 7                | 2.24 |
+  | :white_check_mark: | Debian              | 12 "Bookworm"    | 2.36 |
+  | :white_check_mark: | Fedora              | 38               | 2.37 |
+  | :white_check_mark: | KaOS                |                  | 2.36 |
+  | :white_check_mark: | Mageia              | Cauldron         | 2.36 |
+  | :white_check_mark: | OpenMandriva        | Rolling & Cooker | 2.37 |
+  | :white_check_mark: | openSUSE Tumbleweed |                  | 2.37 |
+  | :x:                | Oracle Linux        | 9                | 2.34 |
+  | :white_check_mark: | PCLinuxOS           |                  | 2.36 |
+  | :x:                | Rocky Linux         | 9                | 2.34 |
+  | :white_check_mark: | Slackware           |                  | 2.37 |
+  | :white_check_mark: | Solus               |                  | 2.36 |
+  | :white_check_mark: | Ubuntu              | 23.04            | 2.35 |
+  | :white_check_mark: | Void Linux          |                  | 2.36 |
+
+  :x: pretty sure not working
+
+  :white_check_mark: should work
+
+confirmed support:
+
+- Ubuntu 23.04
 
 ### Troubleshooting
+
+- may require extra packages on Linux
+
+  ``` bash
+  apt update
+  apt install libdbus-1-3
+  ```
 
 - The `keyring` package auto-detects an appropriate installed keyring backend (see [PyPI - keyring](https://pypi.org/project/keyring/)). Thus may require following additional package if no backend can be detected, see #136
 

